Softeq.XToolkit.Connectivity.iOS/IosConnectivityService.cs

```diff
@@ -0,0 +1,148 @@
+
+        public bool IsSupported => true;
+
+        private void UpdateWiredSnapshot(NWPath nWPath)
+        {
+            HandleUpdateSnapshot(nWPath, NWInterfaceType.Wired);
+        }
+
+        private void UpdateWiFiSnapshot(NWPath nWPath)
+        {
+            HandleUpdateSnapshot(nWPath, NWInterfaceType.Wifi);
+        }
+
+        private void UpdateCeccularSnapshot(NWPath nWPath)
+        {
+            HandleUpdateSnapshot(nWPath, NWInterfaceType.Cellular);
+        }
+
+        private void UpdateLoopbackSnapshot(NWPath nWPath)
+        {
+            HandleUpdateSnapshot(nWPath, NWInterfaceType.Loopback);
+        }
+
+        private void UpdateOtherSnapshot(NWPath nWPath)
+        {
+            HandleUpdateSnapshot(nWPath, NWInterfaceType.Other);
+        }
+
+        private void HandleUpdateSnapshot(NWPath nWPath, NWInterfaceType type)
+        {
+            bool isConnectedOld = IsConnected;
+            var connectionTypesOld = _connectionStatuses.Where(x => x.Value).Select(x => x.Key).ToList();
+
+            _connectionStatuses[type] = nWPath.Status == NWPathStatus.Satisfied;
+
+            if (isConnectedOld != IsConnected)
+            {
+                ConnectivityChanged?.Invoke(this, new ConnectivityChangedEventArgs
+                {
+                    IsConnected = IsConnected
+                });
+            }
+
+            var connectionTypes = _connectionStatuses.Where(x => x.Value).Select(x => x.Key);
+
+            if (!connectionTypes.SequenceEqual(connectionTypesOld))
+            {
+                ConnectivityTypeChanged?.Invoke(this, new ConnectivityTypeChangedEventArgs
+                {
+                    IsConnected = IsConnected,
+                    ConnectionTypes = ConvertTypes(connectionTypes)
+                });
+            }
+        }
+
+        private IEnumerable<ConnectionType> ConvertTypes(IEnumerable<NWInterfaceType> nWInterfaceTypes)
+        {
+            var types = nWInterfaceTypes
+                .Select(x =>
+                {
+                    switch (x)
+                    {
+                        case NWInterfaceType.Cellular:
+                            return ConnectionType.Cellular;
+                        case NWInterfaceType.Wifi:
+                            return ConnectionType.WiFi;
+                        default:
+                            return ConnectionType.Other;
+                    }
+                }).ToList();
+
+            if(types.Contains(ConnectionType.Other) && types.Count > 1)
+            {
+                types.Remove(ConnectionType.Other);
+            }
+
+            return types;
+        }
+
+        private NWPathMonitor CreateMonitor(NWInterfaceType type, Action<NWPath> action)
+        {
+            _connectionStatuses.Add(type, false);
+
+            var monitor = new NWPathMonitor(type);
+            monitor.SetQueue(CoreFoundation.DispatchQueue.MainQueue);
```

Comment:
Avoid using main thread:
`DispatchQueue.GetGlobalQueue(DispatchQueuePriority.Background)`

Reply:
Background thread throws an exception

Reply:
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x10289d2c8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028930b0 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028a1998 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : mono_pmip
	0x21ed069fc - /usr/lib/system/libsystem_platform.dylib : <redacted>
	0x1028921a4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028921a4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1028915c8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x102887140 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101510bcc - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x10122a0f4 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101452560 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101aceec0 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101eaa154 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x10285a26c - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x102859bbc - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x101580e9c - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x1015e7da8 - /var/containers/Bundle/Application/FC8688B3-912E-436D-9B9C-833B72CA6246/Playground.iOS.app/Playground.iOS : (null)
	0x220290eb0 - /usr/lib/libnetwork.dylib : <redacted>
	0x21eb34a38 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eb357d4 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eae6164 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21eae68d4 - /usr/lib/system/libdispatch.dylib : <redacted>
	0x21ed161b4 - /usr/lib/system/libsystem_pthread.dylib : _pthread_wqthread
	0x21ed18cd4 - /usr/lib/system/libsystem_pthread.dylib : start_wqthread

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x10292c49c):0x10292c48c  c0 03 5f d6 ff c3 00 d1 fd 7b 02 a9 fd 83 00 91  .._......{......
0x10292c49c  08 20 40 b9 c8 00 c8 37 88 01 e0 37 00 00 40 f9  . @....7...7..@.
0x10292c4ac  fd 7b 42 a9 ff c3 00 91 c0 03 5f d6 28 10 00 90  .{B......._.(...
0x10292c4bc  08 39 01 91 e8 0b 00 f9 08 10 00 f0 08 4d 3f 91  .9...........M?.

=================================================================
	Managed Stacktrace:
=================================================================
=================================================================